File: go-sdk/pkg/models/operations/getroomsroomid.go
package operations

import (
	"openapi/pkg/models/shared"
)

type GetRoomsRoomIDPathParams struct {
	RoomID string `pathParam:"style=simple,explode=false,name=roomId"`
}

type GetRoomsRoomIDRequest struct {
	PathParams GetRoomsRoomIDPathParams
}

type GetRoomsRoomIDResponse struct {
	ContentType string
	Error       *shared.Error
	Room        *shared.Room
	StatusCode  int64
}

File: typescript-sdk/src/sdk/models/operations/getroomsroomid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetRoomsRoomIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=roomId" })
  roomId: string;
}


export class GetRoomsRoomIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetRoomsRoomIdPathParams;
}


export class GetRoomsRoomIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.ErrorT;

  @SpeakeasyMetadata()
  room?: shared.Room;

  @SpeakeasyMetadata()
  statusCode: number;
}
